### ISSUE TITLE
[C2] Webservices for user progress tracking

### ISSUE THREAD

**Issue by ywarnier** [chamilo/chamilo-lms#5643]
The new API should include:
- a service to get the average learning paths progress of one user in one course/session (-> /users/{id}/courses/{courseId}/stats/avg-lp-progress)
- a service to get the certificates of a user in one course/session (-> /users/{id}/courses/{courseId}/stats/certificates)
- a service to get the global score of a gradebook of a user in one course/session (-> /users/{id}/courses/{courseId}/stats/gradebook-global)
- a service to get the average score in a course/session (across all users) (-> /courses/{id}/stats/course-avg-score)
- a service to get the average progress in a course/session (across all users) (-> /courses/{id}/stats/course-avg-progress)

This is to provide equivalent services to what we have in 1.11

**Comment by christianbeeznest** [comment]
It is added in this PR https://github.com/chamilo/chamilo-lms/pull/6624

Thanks for confirmation.

**Comment by ywarnier** [comment]
Code looks good to me.
Is there a way to add these new services to the /api route generated by ApiPlatform?

**Comment by christianbeeznest** [comment]
Yes — they’re exposed in /api (Swagger UI) as Api Platform Get ops.

GET /api/courses/{id}/stats/{metric}?sessionId=
Metrics: course-avg-score, course-avg-progress

GET /api/users/{id}/courses/{courseId}/stats/{metric}?sessionId=
Metrics: avg-lp-progress, certificates, gradebook-global
